src/OpenTelemetry/Metrics/Metric.cs

```diff
@@ -128,7 +128,10 @@ public sealed class Metric
 
         public MetricPointsAccessor GetMetricPoints()
```

Comment:
We cannot update this method as this would be used by all other exporters for best performance.
We need to add a new method `GetClonedMetricPoints` which should call `this.aggStore.GetClonedMetricPoints`. The user of `InMemoryExporter` has to be aware of this method and call `GetClonedMetricPoints` instead of `GetMetricPoints`:
```
foreach (var metric in batch)
{
    foreach (ref readonly var metricPoint in metric.GetCloneMetricPoints())
    {
        ...
        ...
        ...
    }
}
```